### ISSUE TITLE
Cannot deploy ICP 2.1.0.3 on Ubuntu16.04

### ISSUE THREAD

**Issue by winnietsang** [IBM/deploy-ibm-cloud-private#124]
I try to deploy ICP 2.1.0.3 on 1 master and 3 worker nodes on Ubuntu16.04, however when I deploy it on my environment, I got this error message
TASK [check : Validating Hostname is resolvable] **********************************************************************************
fatal: [xx.xx.xx.xx]: FAILED! => {"changed": false, "msg": "Please configure your hostname to resolve to an externally reachable IP"}
fatal: [xx.xx.xx.xx]: FAILED! => {"changed": false, "msg": "Please configure your hostname to resolve to an externally reachable IP"}
fatal: [xx.xx.xx.xx]: FAILED! => {"changed": false, "msg": "Please configure your hostname to resolve to an externally reachable IP"}
fatal: [xx.xx.xx.xx]: FAILED! => {"changed": false, "msg": "Please configure your hostname to resolve to an externally reachable IP"}

Then I change my hosts file to use hostname instead of ip-address, this time I got this error message
TASK [network-check : Calico Validation - Finding MTU for the detected Interface(s)] **********************************************
fatal: [icp-master]: FAILED! => {"msg": "The conditional check 'hostvars[inventory_hostname]['ansible_'~item]['mtu'] is defined' failed. The error was: error while evaluating conditional (hostvars[inventory_hostname]['ansible_'~item]['mtu'] is defined): 'dict object' has no attribute u'ansible_'\n\nThe error appears to have been in '/installer/playbook/roles/network-check/tasks/calico.yaml': line 86, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n    - name: Calico Validation - Finding MTU for the detected Interface(s)\n      ^ here\n"}
ok: [icp-worker-1] => (item=eth1)
ok: [icp-worker-2] => (item=eth1)
ok: [icp-worker-3] => (item=eth1)
 
I'm not sure what to change to work around this issue. I was able to deploy ICP 2.1.0.3 last week using IP-address on the hosts file, but since this Monday I got the above error message when I try to deploy ICP 2.1.0.3. I even try to deploy older version of ICP like 2.1.0.1 and 2.1.0.2, both get me similar error messages.

**Comment by winnietsang** [comment]
I got it resolve by move my master and worker nodes host-ip to hostname lines to the top of the file instead of append at the end of the file. I can deploy ICP 2.1.0.3 successfully now.

**Comment by theresax** [comment]
I got a similar problem when installing ICP 2.1.0.3 in RHEL 7.5.   See error message below:

^[[0;31mfatal: [10.40.10.2]: FAILED! => {"msg": "The conditional check 'hostvars[inventory_hostname]['ansible_'~item]['mtu'] is defined' failed. The error was: error while evaluating conditional (hostvars[inventory_hostname]['ansible_'~item]['mtu'] is defined): 'dict object' has no attribute u'ansible_'\n\nThe error appears to have been in '/installer/playbook/roles/network-check/tasks/calico.yaml': line 86, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n - name: Calico Validation - Finding MTU for the detected Interface(s)\n ^ here\n"}^[[0m

Switching the order of the entries in the /etc/hosts file didn't help me, but I realized that I was missing the "iproute" on my headNode where I run the ICP installer.   That causes the "ip route ...." commands to fail.  Once I did "yum install iproute", the install was able to complete without issues.

**Comment by penguinpages** [comment]
Following instructions per https://www.youtube.com/watch?v=BURm4C0vg6U  
But on Ubuntu 18 with ICP 2.1.0.3

I added entry for host long and short name top of /etc/hosts
I can ping via DNS forward and reverse
####
administrator@icpdemo01:/opt/ibm-cloud-private-ce-2.1.0.3/cluster$ cat /etc/hosts
172.20.15.252 icpdemo01.ibm.aessatl.arrow.com icpdemo01
127.0.0.1       localhost
127.0.1.1       icpdemo01.ibm.aessatl.arrow.com icpdemo01

# The following lines are desirable for IPv6 capable hosts
::1     ip6-localhost ip6-loopback
fe00::0 ip6-localnet
ff00::0 ip6-mcastprefix
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters
administrator@icpdemo01:/opt/ibm-cloud-private-ce-2.1.0.3/cluster$
administrator@icpdemo01:/opt/ibm-cloud-private-ce-2.1.0.3/cluster$ nslookup
>
> 172.20.12.100
100.12.20.172.in-addr.arpa      name = ibmdc01.ibm.aessatl.arrow.com.

Authoritative answers can be found from:
> 172.20.15.252
252.15.20.172.in-addr.arpa      name = icpdemo01.ibm.aessatl.arrow.com.
252.15.20.172.in-addr.arpa      name = icpdemo01.

Authoritative answers can be found from:
>
###

Still failing on installation:
###
TASK [check : Validating re-install check] *************************************

TASK [check : Copying License files] *******************************************
ok: [172.20.15.252]

TASK [check : Validating CPU cores requirement for single node cluster] ********

TASK [check : Validating memory requirement for single node cluster] ***********

TASK [check : Validating CPU cores requirement for multi node cluster] *********

TASK [check : Validating memory requirement for multi node cluster] ************

TASK [check : Getting DNS server] **********************************************
changed: [172.20.15.252]

TASK [check : Validating DNS server] *******************************************

TASK [check : Checking Hostname is resolvable] *********************************
changed: [172.20.15.252]

TASK [check : Validating Hostname is resolvable] *******************************
fatal: [172.20.15.252]: FAILED! => {"changed": false, "msg": "Please configure your hostname to resolve to an externally reachable IP"}

NO MORE HOSTS LEFT *************************************************************

NO MORE HOSTS LEFT *************************************************************

PLAY RECAP *********************************************************************
172.20.15.252              : ok=26   changed=11   unreachable=0    failed=1

Playbook run took 0 days, 0 hours, 2 minutes, 36 seconds

administrator@icpdemo01:/opt/ibm-cloud-private-ce-2.1.0.3/cluster$
administrator@icpdemo01:/opt/ibm-cloud-private-ce-2.1.0.3/cluster$ ping icpdemo01.ibm.aessatl.arrow.com
PING icpdemo01.ibm.aessatl.arrow.com (127.0.1.1) 56(84) bytes of data.
64 bytes from icpdemo01.ibm.aessatl.arrow.com (127.0.1.1): icmp_seq=1 ttl=64 time=0.070 ms
64 bytes from icpdemo01.ibm.aessatl.arrow.com (127.0.1.1): icmp_seq=2 ttl=64 time=0.061 ms
^C

############


Not sure why local ping returns 127.0.0.1  for FQDN...  but whatever its excuse.. it fails...

Any ideas?  Seems this installer needs a different check method for host name resolution.

**Comment by rkane007** [comment]
Getting the same error on a Mac install.  Any ideas??

T_ASK [network-check : Calico Validation - Finding MTU for the detected Interface(s)] ***
fatal: [192.168.1.108]: FAILED! => 
  msg: |-
    The conditional check 'hostvars[inventory_hostname]['ansible_'~item]['mtu'] is defined' failed. The error was: error while evaluating conditional (hostvars[inventory_hostname]['ansible_'~item]['mtu'] is defined): 'ansible.vars.hostvars.HostVarsVars object' has no attribute u'ansible_'
  
    The error appears to have been in '/installer/playbook/roles/network-check/tasks/calico.yaml': line 82, column 7, but may
    be elsewhere in the file depending on the exact syntax problem.
  
    The offending line appears to be:
  
  
        - name: Calico Validation - Finding MTU for the detected Interface(s)
          ^ here_

**Comment by penguinpages** [comment]
I am not sure if this will be of help but.... I gave up on Ubuntu with IBM ICP for many reasons, first being with the pending RedHat aquistion, I think that will define their development distro :)....  so learn what feeds family. RHEL is best for food..

As for the DNS,  I never did root cause that in that I abandoned Ubuntu, went to RHEL (Centos) and did not have that issue.

But one weird thing I did run into that triggers based on your note is MTU.  I deployed the cluster of VMs on two four servers, total of 9 VMs (3 x master, 3 x proxy, 3 x workers) and had a weird issue where I could not SSH across OVS to other physical servers .. ping worked but not SSH... after some wireshark and sniffer and conversation with IBM support, I enabled 9000MTU on every physical NIC, port (switch), OVS bridge and the VMWare cluster from which I was issuing commands and pushing data etc... and once it was all set to 9000 and I could get 9k pings to work... then SSH was stable.  I know this is NOT a great root cause (could be NIC / Driver / OVS / Physical switch etc... ) number of issues but I was running out of time to setup a class.

Maybe poke that that...
